Dalamud/Game/ClientState/Objects/ObjectTable.cs

```diff
@@ -104,6 +109,20 @@ public unsafe IntPtr GetObjectAddress(int index)
             _ => new GameObject(address),
         };
     }
+
+    /// <inheritdoc />
+    public void Dispose()
+    {
+        this.framework.Update -= this.FrameworkOnUpdate;
+    }
+
+    private void FrameworkOnUpdate(IFramework framework)
```

Comment:
T/N: this *should* be happening before any plugin (since this delegate would get registered very early), but this may cause object table updates to be delayed by a frame depending on what runs before this. I'm unsure if this would matter to Dalamud internal services, but we should probably check.

Reply:
Only place I can see where this would matter would be in `ClientState`. Aside from a circular dependency there on `LocalPlayer`, `ClientState` will register first meaning its framework event gets called prior to the cache update.

This looks like it'd delay login notifications and some other pieces by one frame, though I'm uncertain this would be a notable issue. There _may_ be other uses of `LocalPlayer` that I'm just not immediately seeing.

Would a `.GetDirect(int)` make sense to bypass the cache? I guess we already (indirectly) have it, but it may make sense to make it into a proper internal method for use in cases like LocalPlayer. This will wipe out some of the performance savings introduced by this PR however, as that's a not-insignificant use case.

Reply:
After adding an internal BeforeUpdate this issue should be remedied.